googletest/include/gtest/internal/gtest-internal.h

```diff
@@ -1233,25 +1240,16 @@ struct FlatTupleBase<FlatTuple<T...>, IndexSequence<Idx...>>
 // regardless of T...
 // MakeIndexSequence, on the other hand, it is recursive but with an
 // instantiation depth of O(ln(N)).
-template <typename... T>
-class FlatTuple
-    : private FlatTupleBase<FlatTuple<T...>,
-                            typename MakeIndexSequence<sizeof...(T)>::type> {
-  using Indices = typename FlatTuple::FlatTupleBase::Indices;
 
- public:
-  FlatTuple() = default;
-  explicit FlatTuple(T... t) : FlatTuple::FlatTupleBase(std::move(t)...) {}
-
-  template <size_t I>
-  const typename ElemFromList<I, Indices, T...>::type& Get() const {
-    return static_cast<const FlatTupleElemBase<FlatTuple, I>*>(this)->value;
-  }
+template <typename... Ts>
+struct FlatTuple
+    : FlatTupleBase<typename MakeIndexSequence<sizeof...(Ts)>::type, Ts...> {
+  FlatTuple() {}
 
-  template <size_t I>
-  typename ElemFromList<I, Indices, T...>::type& Get() {
-    return static_cast<FlatTupleElemBase<FlatTuple, I>*>(this)->value;
-  }
+  template <typename... Args>
+  explicit FlatTuple(Args&&... args)
```

Comment:
For a tuple `Tup<int, long, char*>`, this constructor is valid for every possible set of types.

This is the kind of bug that makes me think `std::tuple` is the correct approach.

Reply:
This can be constrained with is_constructable. If you think this is a valid approach I will add static asserts with proper messages to constrain the constructors.

Reply:
Also, further constraints can be applied for the constructor, such as the remove_cv and remove_ref to be the same as the tuple element type, but I'm not sure if that kind of a constraint is necessary.